Dockerfiles/ccp-openshift-scan/scanning/yumupdates.py

```diff
@@ -1,44 +1,117 @@
+        self.needed_fields = ["name", "vra", "repo"]
+
+    def find_updates(self):
+        """
+        Find yum updates and returns the needed_fields attributes of RPM
+        as dictionary for available RPM updates. Returns empty dictionary
+        if no updates are required.
+
+        :return: List of dictionaries of needed_fields as keys and value
+                 as its output
+        :rtype: List
+        """
+        # narrow the package filter to updates
+        rpms = self.yum_obj.doPackageLists(pkgnarrow="updates")
+        updates = []
+        for each in rpms:
+            # check if generator even returned an object
+            # case when no updates are available
+            if not each:
+                continue
+
+            # this is to find out the installed version of
+            # package which is listed in update
+            installed_rpm_gen = self.yum_obj.doPackageLists(
+                pkgnarrow="installed",
+                patterns=[each.name])
+
+            # since it returns a generator, we need to iterate
+            installed_rpm = ""
+            for i in installed_rpm_gen:
+                installed_rpm = i
+                break
+
+            updates.append(
+                {
+                    "name": each.name,
+                    "installed": installed_rpm,
+                    "update": each.ui_nevra,
+                    "repo": each.repo
+                }
+            )
+        return updates
+
+    def print_updates(self, updates):
+        """
+        Print the updates on stdout
+        :arg updates: Updates provided as list of dictionaries,
+                      the output received from find_updates() method
+        :type updates: List
+        """
+        if not updates:
+            print("No RPM updates available as per configured yum repos.")
+            return
+
+        # print the number of updates available
+        print ("About {0} RPM updates are identified.".format(len(updates)))
 
-    out = out.split("\n")
-    print ("RPM\t\t\tNew-Version")
-    for update in out:
-        # split each line separated by tabs
-        update = update.split()
-        print ("{}\t\t{}".format(update[0], update[1]))
+        # now print individual RPM update info
+        for each in updates:
+            # separate updates with extra new lines
+            print("\n")
+            print("Name:        {0}".format(each["name"]))
+            print("Installed:   {0}".format(each["installed"]))
+            print("Update :     {0}".format(each["update"]))
+            print("Yum Repo:    {0}".format(each["repo"]))
 
 
 if __name__ == "__main__":
-    try:
-        find_updates()
-    except Exception as e:
-        print ("Error occurred in RPM Updates scanner execution.")
-        print ("Error: %s".format(e))
-        sys.exit(1)
+    y = YumUpdates()
+    # prevent repo update downloads stdout
+    with SysStdoutSuppressor():
+        updates = y.find_updates()
```

Comment:
The suppressor class is used here, we dont want to print extra output while fetching available updates.